vmc/utils.go

```diff
@@ -90,14 +90,19 @@ func getNSXTReverseProxyURLConnector(nsxtReverseProxyUrl string) (client.Connect
 	return connector, nil
 }
 
-func getTotalSddcHosts(sddc *model.Sddc) int {
-	totalHosts := 0
-	if sddc != nil && sddc.ResourceConfig.Clusters != nil {
+// getHostCountOnPrimaryCluster tries to find the amount of hosts on the primary Cluster in
+// the ResourceConfig of the provided SDDC. If there is no ResourceConfig/Cluster 0 is returned.
+// The primary Cluster is distinguished by its id
+func getHostCountOnPrimaryCluster(sddc *model.Sddc, primaryClusterId string) int {
```

Comment:
nit: since the ID is provided, this method is only getting the number of hosts on the given cluster. It would work for any cluster, right? Sounds weird to have `Primary` in the name here since it seems I could provide the ID of a non-primary cluster to this method and I would get the number of hosts for that cluster?

Reply:
You're correct. I've initially intented for the method to find the primary cluster (without clusterId param) by its name "cluster-X-1", but eventually I found the primary cluster API and decided that addressing a cluster by ID is much better than relying on a implementation detail (like the siffux of the name).